sdk/src/client/secret/mod.rs

```diff
@@ -520,26 +545,33 @@ impl SecretManager {
 pub(crate) async fn default_transaction_unlocks<M: SecretManage>(
     secret_manager: &M,
     prepared_transaction_data: &PreparedTransactionData,
+    protocol_parameters: &ProtocolParameters,
 ) -> crate::client::Result<Unlocks>
 where
     crate::client::Error: From<M::Error>,
 {
     let transaction_signing_hash = prepared_transaction_data.transaction.signing_hash();
     let mut blocks = Vec::new();
     let mut block_indexes = HashMap::<Address, usize>::new();
-    let slot_index = prepared_transaction_data.transaction.creation_slot();
+    let slot_index = prepared_transaction_data
+        .transaction
+        .context_inputs()
+        .iter()
+        .find_map(|c| c.as_commitment_opt().map(|c| c.slot_index()))
+        .unwrap_or_else(|| prepared_transaction_data.transaction.creation_slot());
```

Comment:
Is that a TIP thing that we should favour commitment and take creation slot if none is provided ?

Reply:
No, do you prefer `unwrap_or(0)`? Having it optional is a bit annoying and if there is no commitment input then it shouldn't be used anyways